nri-http/src/Http.hs

```diff
@@ -212,36 +220,102 @@ _request doAnythingHandler manager settings = do
                       |> HTTP.responseTimeoutMicro
                 }
         HTTP.httpLbs finalRequest requestManager
-    pure <| case response of
-      Right okResponse ->
-        case decode (Internal.expect settings) (HTTP.responseBody okResponse) of
-          Ok decodedBody ->
-            Ok decodedBody
-          Err message ->
-            Err (Internal.BadBody message)
-      Left (HTTP.HttpExceptionRequest _ content) ->
-        case content of
-          HTTP.StatusCodeException res _ ->
-            let statusCode = fromIntegral << Status.statusCode << HTTP.responseStatus
-             in Err (Internal.BadStatus (statusCode res))
-          HTTP.ResponseTimeout ->
-            Err Internal.Timeout
-          HTTP.ConnectionTimeout ->
-            Err (Internal.NetworkError "ConnectionTimeout")
-          HTTP.ConnectionFailure err ->
-            Err (Internal.NetworkError (Text.fromList (Exception.displayException err)))
-          err ->
-            Err (Internal.NetworkError (Text.fromList (show err)))
-      Left (HTTP.InvalidUrlException _ message) ->
-        Err (Internal.BadUrl (Text.fromList message))
-
-decode :: Expect a -> Data.ByteString.Lazy.ByteString -> Result Text a
-decode Internal.ExpectJson bytes =
-  case Aeson.eitherDecode bytes of
-    Left err -> Err (Text.fromList err)
-    Right x -> Ok x
-decode Internal.ExpectText bytes = (Ok << Data.Text.Lazy.toStrict << Data.Text.Lazy.Encoding.decodeUtf8) bytes
-decode Internal.ExpectWhatever _ = Ok ()
+    pure <| handleResponse (Internal.expect settings) response
+
+handleResponse :: Expect a -> Either HTTP.HttpException (HTTP.Response Data.ByteString.Lazy.ByteString) -> Result Error a
+handleResponse expect response =
+  case response of
+    Right okResponse ->
+      let bytes = HTTP.responseBody okResponse
+          bodyAsText = Data.Text.Lazy.toStrict <| Data.Text.Lazy.Encoding.decodeUtf8 bytes
+       in case expect of
+            Internal.ExpectJson ->
+              case Aeson.eitherDecode bytes of
+                Left err -> Err (Internal.BadBody (Text.fromList err))
+                Right x -> Ok x
+            Internal.ExpectText -> Ok bodyAsText
```

Comment:
Should we use https://hackage.haskell.org/package/text-2.0.1/docs/Data-Text-Encoding.html#v:decodeUtf8-39-
here?

Reply:
maybe everywhere where we docedUtf8 rn?